lightning-invoice/src/payment.rs

```diff
@@ -721,7 +744,26 @@ mod tests {
 
 		let payment_preimage = PaymentPreimage([1; 32]);
 		let invoice = expired_invoice(payment_preimage);
+		if let PaymentError::Invoice(msg) = invoice_payer.pay_invoice(&invoice).unwrap_err() {
+			assert_eq!(msg, "Invoice expired prior to send");
+		} else { panic!("Expected Invoice Error"); }
+	}
+
+	#[test]
+	fn fails_retrying_invoice_after_expiration() {
+		let event_handled = core::cell::RefCell::new(false);
+		let event_handler = |_: &_| { *event_handled.borrow_mut() = true; };
+
+		let payer = TestPayer::new();
+		let router = TestRouter {};
+		let logger = TestLogger::new();
+		let invoice_payer =
+			InvoicePayer::new(&payer, router, &logger, event_handler, RetryAttempts(2));
+
+		let payment_preimage = PaymentPreimage([1; 32]);
+		let invoice = will_expire_in_1_sec_invoice(payment_preimage);
 		let payment_id = Some(invoice_payer.pay_invoice(&invoice).unwrap());
+		std::thread::sleep(Duration::from_secs(2));
```

Comment:
Hmm? The point of this test is that, unlike `fails_paying_invoice_after_expiration`, we expire while waiting to retry instead of expiring before the original `pay_invoice` call.

Reply:
Yeah, but as far as `InvoicePayer` is concerned, it doesn't care what the original `Invoice` said the expiry was when retrying. It just cares what's in the `PaymentPathFailed` event that the test is later feeding it. So that can be changed to have expired, thus simulating the passage of time.

Reply:
Ah, that suggestion was too clever by half.

Reply:
Ah, that suggestion was too clever by half. Did that now.